### PR TITLE
Implements GET /notes route for pre and post

### LINKED ISSUE #63: Implement GET /notes route
**Requirements for feature**
- [ ] When GET is sent on URL/notes/<id>, respond with the questions and their types
- [ ] Might be useful to return the question ID as well 

**Any context for feature**
This route will be called before setting up the pre/post session forms

**Resources / tips for development**
- Might be useful to return the ID of the question in this request. That way when the frontend sends the PATCH notes request, it can just send the ID instead of the question text. 
- This will be especially handy when we move our questions to mongoDB

### PR DESCRIPTION
**Pull request by christen03**
### Tracking Info

Resolves #63 

### Changes

`routes/notes.ts` - I made a GET route that returns answer data from pre/post session forms in the desired format. 
`src/app.ts` - Included the new notes router into the app.

### Testing

I ran the testing with Postman, when I send a GET request on Postman with the in the format: `localhost:3000/notes/<id>`
Where `<id>` is the ObjectID of the MongoDB document of the note we are trying to receive, I get this response: 

image

If the ID does not exist, it fails and returns a not found:

image

### Confirmation of Change 

Changes listed above. Look in `routes/notes.ts` for the main change.

### CODE REVIEW

File: backend/src/routes/notes.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+router.get("/notes/:id", async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    console.log("Getting...");
+    const id = req.params.id;
+    const note = await Note.findById(id);
+    if (note == null) {
+      throw new Error();
+    }
+    return res.status(200).json(note.answers);
```
Comment (AmanKAggarwal) [inline backend/src/routes/notes.ts:14]: We need to send the question in this GET response as well. The frontend will not hardcode the questions but will rely on what comes from the backend

**Review by wllmwu — COMMENTED** [review]
LGTM

File: backend/src/routes/notes.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  } catch (e) {
+    next(e);
```
Comment (adhi0331) [inline backend/src/routes/notes.ts:16]: Just letting you know that next(e) isn't actually doing anything yet cause we don't have a error handler function and we've just been coding the handlers in the catch block itself. Although we should prolly at some point create an error handler function that is used for all routes.

File: backend/src/routes/notes.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import express, { NextFunction, Request, Response } from "express";
+import { Note } from "../models/notes";
+
+const router = express.Router();
+
+router.get("/notes/:id", async (req: Request, res: Response, next: NextFunction) => {
```
Comment (adhi0331) [inline backend/src/routes/notes.ts:6]: It's good for now but once the get user PR gets approved be sure to add the authorization middleware. These routes should be user protected.

**Review by adhi0331 — APPROVED** [review]
LGTM 🚀

File: backend/src/routes/notes.ts (hunk 1 above)
Comment (AmanKAggarwal) [reply backend/src/routes/notes.ts:14]: @christen03 lets add this and then we should be good to merge this

**Comment by AmanKAggarwal** [comment]
Working screenshot
image

**Review by AmanKAggarwal — APPROVED** [review]
LGTM

**Pull request merged**